### PR TITLE
Added menu bar

### PR DESCRIPTION
**Pull request by Lei-Tin**
我真尽力了，this thing is so hard :CCCCCC
Spent like 5 hours on this and 除了 crash 之外什么都没得到哈哈哈哈哈哈

Added menu bar, some functions to it
Added an About window, looks bad at the moment
Added a settings window, currently empty

Fixed a very tiny wording error in the application
Label changes after you initialized the data at least once
Plot now starts with default labels

Only selected some tools to keep in the Navigation Toolbar
Made Windows recognize our application and added an icon

### CODE REVIEW

File: Application/gui_utils.py [hunk 1]
```diff
@@ -9,11 +9,11 @@
 from PyQt5.QtCore import *
 from PyQt5.QtGui import *
 from PyQt5.QtWidgets import *
+from matplotlib.backends.backend_qt5agg import NavigationToolbar2QT as NavigationToolbar
```
Comment (ScottCTD) [inline Application/gui_utils.py:12]: We should separate the import statements into groups and with a comment

More organized

File: Application/gui_utils.py [hunk 2]
```diff
@@ -189,3 +189,36 @@ class StandardLineEdit(QLineEdit):
...
+class StandardMenuBar(QMenuBar):
+    """
+    A standard menu bar for our project.
+
+    When needed, we could add more attributes and methods.
+    """
+
+    def __init__(self, parent: Optional[QWidget] = None) -> None:
+        super(StandardMenuBar, self).__init__(parent)
+        set_font(self)
```
Comment (ScottCTD) [inline Application/gui_utils.py:203]: menubar的字体貌似有点大，太傻了感觉

File: Application/gui_utils.py [hunk 3]
```diff
@@ -189,3 +189,36 @@ class StandardLineEdit(QLineEdit):
...
+
+
+class StandardNavigationToolbar(NavigationToolbar):
+    """
+    A standard menu for our project.
+
+    When needed, we could add more attributes and methods.
+    """
+    toolitems = [t for t in NavigationToolbar.toolitems if t[0] in ('Home', 'Pan', 'Zoom')]
```
Comment (ScottCTD) [inline Application/gui_utils.py:224]: Goodddd

File: Application/gui_main.py [hunk 4]
```diff
@@ -14,16 +14,21 @@
...
 import resource_manager
 from gui_utils import *
 
-# PyQt5
+# Ctype
+import ctypes
+myappid = 'CSC110.covid_school_plot'  # Random identifier
+
+# Letting Windows display the Icon in the taskbar as well
+ctypes.windll.shell32.SetCurrentProcessExplicitAppUserModelID(myappid)
```
Comment (ScottCTD) [inline Application/gui_main.py:29]: GOOOD

File: Application/gui_main.py [hunk 5]
```diff
@@ -735,3 +854,31 @@ def on_end_date_slider_moved(self, new_value: int) -> None:
...
+        """Saves the current plots at the specified location"""
+        # The second return value is "Selected filter", which is useless
+        path, _ = QFileDialog.getSaveFileName(self, "Save Image", "", "PNG(*.png)")
+
+        if path == '':
+            # This is because the user may press cancel
+            return
+
+        # Saving canvas at desired path
+        self.plot_canvas.print_png(path)
```
Comment (ScottCTD) [inline Application/gui_main.py:868]: GOOOD

File: Application/gui_main.py [hunk 6]
```diff
@@ -184,6 +196,63 @@ def on_mouse_move(self, event: matplotlib.backend_bases.MouseEvent) -> None:
                     self.draw()
 
 
+class Settings(QWidget):
```
Comment (ScottCTD) [inline Application/gui_main.py:199]: As we discussed in wechat

better not add additional windows

sry我说的比较直接可能没有很让你开心 因为毕竟是你写出的代码

但是我比较可恶，所以还是得说 内容太多window确实不错 但是咱内容不太多 咱的那种setting类似pycharm放menu bar就挺好

File: Application/gui_main.py [hunk 7]
```diff
@@ -462,12 +551,41 @@ def __init__(self, *args, **kwargs) -> None:
...
         self.init_signals()
 
         # Now, our data haven't initialized yet, so we disable all functional widgets for safety.
         self.set_enabled_functional_widgets(False)
 
+    def init_menu(self) -> None:
+        """
+        Initialize the menu bar
+        """
+        save_plot = QAction('Save current plot', self)
```
Comment (ScottCTD) [inline Application/gui_main.py:567]: We Should Use Title Case

**Review by ScottCTD — CHANGES_REQUESTED** [review]
Some small fixes

I suggest completely finishing functionality and then pr, instead of leaving two blank windows lol

File: Application/gui_main.py [hunk 8]
```diff
@@ -14,16 +14,21 @@
...
 
-# PyQt5
+# Ctype
+import ctypes
+myappid = 'CSC110.covid_school_plot'  # Random identifier
+
+# Letting Windows display the Icon in the taskbar as well
+ctypes.windll.shell32.SetCurrentProcessExplicitAppUserModelID(myappid)
 
+# PyQt5
```
[the remark this answers is not in the document]
Comment (Lei-Tin) [reply Application/gui_main.py:31]: 删了

File: Application/gui_main.py [hunk 9]
```diff
@@ -14,16 +14,21 @@
...
 # Our modules
 import algorithms
 import data
 import resource_manager
 from gui_utils import *
 
-# PyQt5
+# Ctype
+import ctypes
+myappid = 'CSC110.covid_school_plot'  # Random identifier
```
[the remark this answers is not in the document]
Comment (Lei-Tin) [reply Application/gui_main.py:26]: image
Like this 吗

File: Application/gui_utils.py (hunk 1 above)
Comment (Lei-Tin) [reply Application/gui_utils.py:12]: Alright

File: Application/gui_utils.py (hunk 3 above)
Comment (Lei-Tin) [reply Application/gui_utils.py:224]: GOOOOOOODDDDD

File: Application/gui_utils.py (hunk 2 above)
Comment (Lei-Tin) [reply Application/gui_utils.py:203]: 改成 12pt 您意下如何

File: Application/gui_main.py (hunk 5 above)
Comment (Lei-Tin) [reply Application/gui_main.py:868]: GOOOOOOOODDDDDDDD

File: Application/gui_main.py (hunk 4 above)
Comment (Lei-Tin) [reply Application/gui_main.py:29]: GOOOOOOOOOOOOD

File: Application/gui_main.py (hunk 6 above)
Comment (Lei-Tin) [reply Application/gui_main.py:199]: No it's ok, im 不知道要加什么在 menu bar 所以才加上这个功能的哈哈哈哈哈哈

File: Application/gui_main.py (hunk 7 above)
Comment (Lei-Tin) [reply Application/gui_main.py:567]: you mean savePlot? wat

Comment (ScottCTD) [reply Application/gui_main.py:567]: Same Current Plot

File: Application/gui_utils.py (hunk 2 above)
Comment (ScottCTD) [reply Application/gui_utils.py:203]: 不是 别您 太搞了

你打开软件看看 对比一波pycharm的那种就行

我没记错的咱的默认就是12

File: Application/gui_main.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (ScottCTD) [reply Application/gui_main.py:26]: yes I thnik os

File: Application/gui_utils.py (hunk 2 above)
Comment (Lei-Tin) [reply Application/gui_utils.py]: 默认是 14，现在改成 12 了
请您查阅

File: Application/gui_main.py (hunk 7 above)
Comment (Lei-Tin) [reply Application/gui_main.py]: Alright

**Review by CharlotteChenyy — APPROVED** [review]
Greattt!!

**Comment by ScottCTD** [comment]
这也是屎山代码

慢慢改 别merge

**Review by ScottCTD — APPROVED** [review]

**Pull request merged**